youtube2zim/scraper.py

```diff
@@ -572,71 +578,69 @@ def get_slot():
+            if not self.s3_storage.has_object(key, self.s3_storage.bucket_name):
+                return False
+        else:
+            if not self.s3_storage.has_object_matching_meta(
+                key, tag="encoder_version", value=ENCODER_VERSION
+            ):
+                return False
+        video_path.parent.mkdir(parents=True, exist_ok=True)
         try:
-            if self.use_any_optimized_version:
-                if not self.s3_storage.has_object(s3_key, self.bucket_name):
-                    return False
-            else:
-                if not self.s3_storage.has_object_matching_meta(
-                    s3_key,
-                    tag="encoding_version",
-                    value=ENCODER_VERSION,
-                    bucket_name=self.bucket_name,
-                ):
-                    return False
-            os.makedirs(os.path.dirname(local_video_path), exist_ok=True)
-            self.s3_storage.download_file(s3_key, local_video_path)
+            self.s3_storage.download_file(key, video_path)
+            logger.info(f"cache ({key}) downloaded {video_path}")
+            return True
         except Exception as ex:
-            logger.error(f"{self.bucket_name}/{s3_key} download failed with {str(ex)}")
+            logger.error(f"{key} download failed with {ex}")
             return False
-        logger.info(
-            f"cache ({self.bucket_name}/{s3_key}) downloaded {local_video_path}"
-        )
-        return True
 
-    def upload_s3_cache(self, s3_key, local_video_path):
-        """returns True if video uploaded to S3 for cache otherwise None"""
+    def upload_to_cache(self, key, video_path):
+        """ whether it successfully uploaded to cache """
         try:
             self.s3_storage.upload_file(
-                local_video_path, s3_key, meta={"encoding_version": ENCODER_VERSION}
+                video_path, key, meta={"encoder_version": ENCODER_VERSION}
             )
-            logger.info(f"cached video to {self.bucket_name} bucket ({s3_key})")
+            logger.info(f"video cached to {key}")
             return True
         except Exception as ex:
-            logger.error(f"{self.bucket_name}/{s3_key} upload failed with {str(ex)}")
+            logger.error(f"{key} upload failed with {ex}")
+            return False
 
     def download_video_files_batch(self, options, videos_ids):
         succeeded = []
         failed = []
+        # Lets make copy of options dictionary so that it is not shared across concurrent events
+        options_copy = options.copy()
         for video_id in videos_ids:
-            video_location = options["y2z_videos_dir"].joinpath(video_id)
+            video_location = options_copy["y2z_videos_dir"].joinpath(video_id)
             # Reset the skip_download argument
-            options["skip_download"] = False
-            cache_present = None
-            if self.s3_url_with_credentials:
+            options_copy["skip_download"] = False
```

Comment:
Now that we have a copy, could we have a copy per video? It would have no memory impact since it's in a loop but it's cleaner and we only set `skip_download` when we need it.

Reply:
Ok. That will remove another line. Done.